src/node.js

```diff
@@ -717,6 +717,12 @@
         stdin._handle.readStop();
       });
 
+      stdin.isTTY = true;
+      stdin.isRaw = false;
+      stdin.setRawMode = function setRawMode() {
+        throw new Error('Not a raw device');
+      };
```

Comment:
This looks incorrect to me.  `.isTTY` should only be true when stdin refers to a real tty, not when it's a file/pipe/socket/etc.

You could wrap it in a `if (!process.stdin.hasOwnProperty('isTTY')) { ... }` block, although that feels kind of icky (and the fact that `.isTTY` is an own property is something of an implementation detail, so `if (!('isTTY' in process.stdin)) { .. }` is probably more correct.)